controllers/metal3machine_controller.go

```diff
@@ -247,7 +249,9 @@ func (r *Metal3MachineReconciler) reconcileNormal(ctx context.Context,
 			)
 		}
 		if bmhID != nil {
-			providerID = fmt.Sprintf("metal3://%s", *bmhID)
+			provID := fmt.Sprintf("metal3.io/%s-%s", *bmhID, m3mID)
+			hmdprovIDHash := md5.Sum([]byte(provID))
+			providerID = fmt.Sprintf("%x", hmdprovIDHash)
```

Comment:
isn't it so that we should have the `metal3://` prefix ? I remember from some log that absent this prefix, CAPI controllers become unhappy

Reply:
There are two versions.
1. when setting the providerID
2. When calling SetNodeProviderID